### ISSUE TITLE
Ability to run builds within Lambda now that it supports Docker

### ISSUE THREAD

**Issue by jakejscott** [buildkite/elastic-ci-stack-for-aws#777]
Would be pretty cool to be able to run a build within Lambda :-) 

Wondering if it would be possible?

Would allow super cheap and fast builds (that can run within 15 minutes)

**Comment by yob** [comment]
I haven't looked into the Lambda+Docker announcement in detail yet, but I don't see why it couldn't be used to run a container with the agent.

We'd love to see it happen! However, I'm not sure if it's a good fit for the Elastic Stack. Mainly because the elastic stack strongly encourages a certain shaped build (minimal tooling, docker, build dependencies declared in the code repo, etc) and any builds run on lambda would necessarily look quite different (I assume there's no way to run docker-in-docker?).

One open question: would we need something to orchestrate (eg. start) the lambda agents, and would it be a new lambda per job? If so, I assume the `--acquire-job` flag would be involved.
 
tldr; we're keen, but probably built somewhere outside this repo :heart:

**Comment by lox** [comment]
This would be a great fit for https://github.com/buildkite/elastic-ci-stack-for-aws-codebuild

**Comment by jakejscott** [comment]
@lox @yob Thanks for the replies :) Yeah the docker in docker thing is weird... I guess I meant you could supply your own docker container that has the dependencies you want already baked into it for the build + the buildkite agent...then you'd just need some way to trigger a lambda to spin up and start a build.

I've used codebuild a bit and find it's a bit of a hassle to configure each pipeline using Terraform/Cloudformation/CDK etc for each thing you wanna build (plus some pain around hooking that up with github).

Would be nice not having a VPC and security to worry about.
